cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -455,66 +477,65 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
      * @param kafkaRebalance Current {@code KafkaRebalance} resource
      * @param rebalanceAnnotation The current value for the strimzi.io/rebalance annotation
      * @param rebalanceOptionsBuilder builder for the Cruise Control REST API client options
-     * @return a Future with the next {@code KafkaRebalanceStatus} bringing the state
+     * @return a Future with the next {@code KafkaRebalanceStatus} including the state
      */
     private Future<KafkaRebalanceStatus> onPendingProposal(Reconciliation reconciliation,
                                                            String host, CruiseControlApi apiClient,
                                                            KafkaRebalance kafkaRebalance,
                                                            KafkaRebalanceAnnotation rebalanceAnnotation,
                                                            RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
         Promise<KafkaRebalanceStatus> p = Promise.promise();
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.none) {
-            log.debug("{}: Arming Cruise Control rebalance proposal request timer", reconciliation);
-            vertx.setPeriodic(REBALANCE_POLLING_TIMER_MS, t -> {
-                kafkaRebalanceOperator.getAsync(kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName()).onComplete(getResult -> {
-                    if (getResult.succeeded()) {
-                        KafkaRebalance freshKafkaRebalance = getResult.result();
-                        // checking that the resource wasn't delete meanwhile the timer wasn't raised
-                        if (freshKafkaRebalance != null) {
-                            // checking it is in the right state because the timer could be called again (from a delayed timer firing)
-                            // and the previous execution set the status and completed the future
-                            if (state(freshKafkaRebalance) == KafkaRebalanceState.PendingProposal) {
-                                if (rebalanceAnnotation(freshKafkaRebalance) == KafkaRebalanceAnnotation.stop) {
+        if (KafkaRebalanceAnnotation.none.equals(rebalanceAnnotation)) {
```

Comment:
I agree with @ppatierno, that the "==" version reads cleaner that this ".equals" version